e2e/test/scenarios/visualizations-charts/pie_chart.cy.spec.js

```diff
@@ -106,14 +106,36 @@ describe("scenarios > visualizations > pie chart", () => {
       dataset_query: testQuery,
       display: "pie",
     });
+    cy.findByTestId("chart-container").within(() => {
```

Comment:
Is this the actual fix? To wait until the chart is ready before we start to assert on the legend items? (just trying to understand the change)

Reply:
The key is line 110, which ensures that the page is properly ready to be interacted with.